modeling/local-slice-algorithm.go

```diff
@@ -146,7 +146,8 @@ func balanceSliceGroups(endpointsAvailable *endpointsList, endpointsNeeded *endp
 		// in this case, we assign those extra endpoints to its local
 		// endpointSliceGroups
 		for _, extraEndpoints := range endpointsAvailable.byZone {
-			sliceGroups[extraEndpoints.name].Composition[extraEndpoints.name] += extraEndpoints.deviation
+			originalEndpoints := sliceGroups[extraEndpoints.name].Composition[extraEndpoints.name]
+			sliceGroups[extraEndpoints.name].Composition[extraEndpoints.name] = weightedEndpoints{Number: originalEndpoints.Number + extraEndpoints.deviation, Weight: originalEndpoints.Weight}
```

Comment:
This is a long line, maybe try splitting it up a bit:
```suggestion
			sliceGroups[extraEndpoints.name].Composition[extraEndpoints.name] = weightedEndpoints{
				Number: originalEndpoints.Number + extraEndpoints.deviation,
				Weight: originalEndpoints.Weight,
			}
```

Reply:
changed